### PR TITLE
Partial indexes support added!

### PR DESCRIPTION
**Pull request by yashdeeph709**
@SmofIndex(fields = {
            @SmofIndexField(name = "name", type = IndexType.TEXT), 
            @SmofIndexField(name = "creator", type = IndexType.TEXT)
},
pfe=@SmofPFEQuery(
            expression=@SmofQueryA(
                      name = "name",
                      query={
                                @SmofFilter(
                                          operator = OperatorType.exists, 
                                          value = "true"
                                 )
                      }
            )
))
For Adding partial Index to an entity you can add a SmofPFEQuery which takes name of the field and the operator and value for the partial filter
{ "partialFilterExpression" : { "name" : { "exists":true}} }
The above annotations are for this document for partial filter.

### CODE REVIEW

File: src/main/java/org/smof/annnotations/OperatorType.java [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+    */
+    greaterThanEquals("$gte"),
+    /**
+    * $lte represents lessThanEquals.
+    */
+    lessThanEquals("$lte"),
+    /**
+    * $exists represents exists.
+    */
+    exists("exists");
```
Comment (JPDSousa) [inline src/main/java/org/smof/annnotations/OperatorType.java]: Is it `exists` or `$exists`?  (missing `$`)

File: src/main/java/org/smof/annnotations/SmofPFEQuery.java [hunk 2]
```diff
@@ -0,0 +1,17 @@
+package org.smof.annnotations;
```
Comment (JPDSousa) [inline src/main/java/org/smof/annnotations/SmofPFEQuery.java:1]: This annotation is supposed to be a wrapper around `@SmofQueryA` right? If so, why is it needed?

File: src/main/java/org/smof/annnotations/SmofFilter.java [hunk 3]
```diff
@@ -0,0 +1,16 @@
+package org.smof.annnotations;
```
Comment (JPDSousa) [inline src/main/java/org/smof/annnotations/SmofFilter.java:1]: Awesome 👍

File: src/main/java/org/smof/annnotations/SmofPartialIndex.java [hunk 4]
```diff
@@ -0,0 +1,17 @@
+package org.smof.annnotations;
```
Comment (JPDSousa) [inline src/main/java/org/smof/annnotations/SmofPartialIndex.java]: Is this annotation still needed?

File: src/main/java/org/smof/annnotations/SmofQueryA.java [hunk 5]
```diff
@@ -0,0 +1,16 @@
+package org.smof.annnotations;
```
Comment (JPDSousa) [inline src/main/java/org/smof/annnotations/SmofQueryA.java]: Exactly what i was planing to do 😉 !

**Comment by JPDSousa** [comment]
@yashdeeph709  I think a cool way to test this would be to load the index data and check if such data is valid according to the index expression.

I'm gonna check the java-driver documentation to see if such think is possible. I can help you with this if you want to.

**Comment by yashdeeph709** [comment]
thanks jp, I will improve on it tomorrow.

File: src/main/java/org/smof/annnotations/SmofPartialIndex.java (hunk 4 above)
Comment (yashdeeph709) [reply src/main/java/org/smof/annnotations/SmofPartialIndex.java]: No we dont need that I will delete that and improve the pr.

File: src/main/java/org/smof/annnotations/OperatorType.java (hunk 1 above)
Comment (yashdeeph709) [reply src/main/java/org/smof/annnotations/OperatorType.java]: yeah I mistaken

**Comment by JPDSousa** [comment]
Ok @yashdeeph709 , i think that's all! I'll accept the pull request, and i'll then run additional tests and try this feature in other projects of mine where i use Smof. Thanks a lot for you contribution. You did an amazing job!! 👍 👍 

Feel free to contribute to any other issues!

**Comment by yashdeeph709** [comment]
It was pleasure to work with you. Feeling happy to get my first pull request accepted. I will continue my work here. :smile:

**Pull request merged**